extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/KafkaIndexTaskClient.java

```diff
@@ -76,33 +83,82 @@ public TaskNotRunnableException(String message)
   private final HttpClient httpClient;
   private final ObjectMapper jsonMapper;
   private final TaskInfoProvider taskInfoProvider;
+  private final Duration httpTimeout;
   private final RetryPolicyFactory retryPolicyFactory;
-
-  public KafkaIndexTaskClient(HttpClient httpClient, ObjectMapper jsonMapper, TaskInfoProvider taskInfoProvider)
+  private final ListeningExecutorService executorService;
+  private final long numRetries;
+
+  public KafkaIndexTaskClient(
+      HttpClient httpClient,
+      ObjectMapper jsonMapper,
+      TaskInfoProvider taskInfoProvider,
+      String dataSource,
+      int numThreads,
+      Duration httpTimeout,
+      long numRetries
+  )
   {
     this.httpClient = httpClient;
     this.jsonMapper = jsonMapper;
     this.taskInfoProvider = taskInfoProvider;
+    this.httpTimeout = httpTimeout;
+    this.numRetries = numRetries;
     this.retryPolicyFactory = createRetryPolicyFactory();
+
+    String threadId = String.format("KafkaIndexTaskClient-%s", dataSource);
```

Comment:
`String.format("KafkaIndexTaskClient-%s-%%d", dataSource)` should work too, although changing is unnecessary